sycl/source/detail/config.hpp

```diff
@@ -378,6 +378,64 @@ template <> class SYCLConfig<SYCL_QUEUE_THREAD_POOL_SIZE> {
   }
 };
 
+static size_t ParseReductionMaxWG(const char *ValueStr,
+                                  const char *ConfigName) {
+  // Default to 0 to signify an unset value.
+  if (!ValueStr)
+    return 0;
+
+  int Result = 1;
+  try {
+    Result = std::stoi(ValueStr);
+  } catch (...) {
+    throw sycl::exception(
+        sycl::make_error_code(sycl::errc::invalid),
+        "Invalid value for " + std::string{ConfigName} +
+            " environment variable: value should be a number");
+  }
+
+  if (Result < 1)
+    throw sycl::exception(
+        sycl::make_error_code(sycl::errc::invalid),
+        "Invalid value for " + std::string{ConfigName} +
+            " environment variable: value should be larger than zero");
+
+  return Result;
+}
+
+template <> class SYCLConfig<SYCL_REDUCTION_MAX_WORKGROUP_SIZE_CPU> {
```

Comment:
3 classes  duplicated and having only different input val...
I would think about having 1 common/base class for them, but if you decide not changing it right now, I'm Ok with the fix as well.

Reply:
BTW, thank you for doing these changes!

Reply:
Good call. I have changed it to use a common superclass.